src/parser.ts

```diff
@@ -0,0 +1,39 @@
+interface Token {
```

Comment:
Do I want interface here, or type?

Reply:
Based on how it's used here, Token should be a type.

But in the future, if you want to merge definitions, keep it an interface